packages/server/graphql/public/mutations/updateTemplateCategory.ts

```diff
@@ -0,0 +1,38 @@
+import {SubscriptionChannel} from 'parabol-client/types/constEnums'
+import getKysely from '../../../postgres/getKysely'
+import publish from '../../../utils/publish'
+import {MutationResolvers} from '../resolverTypes'
+
+const updateTemplateCategory: MutationResolvers['updateTemplateCategory'] = async (
+  _source,
+  {templateId, mainCategory},
+  {dataLoader, socketId: mutatorId}
+) => {
+  const operationId = dataLoader.share()
+  const subOptions = {mutatorId, operationId}
+  const pg = getKysely()
+
+  // VALIDATION
+  if (mainCategory.length < 1) return {error: {message: 'Category name cannot be blank'}}
+  if (mainCategory.length > 100) return {error: {message: 'Category name is too long'}}
+
+  // I'd like to relax this later, hence the separate validation from string length
```

Comment:
-1 Let's just add a TODO for the string length validation to make this simpler, since we don't yet have plans for custom categories.

Even for custom categories, we'd probably have a separate "create category" mutation, such that we'd just take a category ID here.

Reply:
will do